api/filters/filtersutil/setters.go

```diff
@@ -8,9 +8,16 @@ import (
 type SetFn func(*yaml.RNode) error
 
 // SetScalar returns a SetFn to set a scalar value
-func SetScalar(value string) SetFn {
+func SetScalar(value, tag string) SetFn {
```

Comment:
I'm wondering if it's okay to pass key to `SetScalar` delete `SetEntry` since this can be passed to `TrackableSetter.SetScalar` with empty key val just fine, and I don't think this is used outside of `api/filters`

Reply:
Is this necessary for your PR to work?

We can, if there is no other feasible implementation, but this is a public function in a library used by consumers outside of kustomize, so I would prefer if we can minimize these kinds of changes.

Reply:
Ah, I see your longer comment above with more details. I will take a closer look a bit later at your approach.